source/base/ThreadPool.ooc

```diff
@@ -70,7 +76,15 @@ _TaskPromise: class extends Promise {
 		this _task free()
 		super()
 	}
-	wait: override func -> Bool { this _task wait() }
+	wait: func -> Bool { this _task wait() }
+	wait: func ~timeout (seconds: Double) -> Bool {
+		timer := ClockTimer new() . start()
+		status := false
+		while (timer stop() / 1000.0 < seconds && !status)
```

Comment:
Don't you need to call `timer start()` again after you call `timer stop()`? Or does it simply return the current time difference since the latest call to `timer start()`?

Nonetheless, this is a very ugly solution to wait for something (as I think you already know by now). At least sleep for `timeout / 10` or so between every check.